lix/packages/sdk/src/database/applySchema.ts

```diff
@@ -37,7 +36,15 @@ export async function applySchema(args: { sqlite: SqliteDatabase }) {
     created_at TEXT DEFAULT CURRENT_TIMESTAMP NOT NULL
   ) strict;
 
-  CREATE INDEX IF NOT EXISTS idx_change_parent_id ON change (parent_id);
+  CREATE TABLE IF NOT EXISTS change_edge (
+    parent_id TEXT NOT NULL,
+    child_id TEXT NOT NULL,
+    PRIMARY KEY (parent_id, child_id),
+    FOREIGN KEY(parent_id) REFERENCES change(id),
+    FOREIGN KEY(child_id) REFERENCES change(id),
+    -- Prevent self referencing edges
+    CHECK (parent_id != child_id)
+  ) strict;
```

Comment:
We should add an index to the child id as well. I think the parent_id should be covered by the compound primary key - but child_id will most likely need a full table scan in this schema.